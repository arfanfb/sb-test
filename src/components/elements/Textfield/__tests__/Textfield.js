import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TextField from '../index';

describe('src/components/elements/textfield', () => {
  it('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<TextField />);
    expect(tree).toMatchSnapshot();
  });

  it('expect width 400px', () => {
    const result = TextField({
      ...TextField.defaultProps,
      width: 400,
    });

    const { width } = result.props;
    expect(width).toBe(400);
  });

  it('expect class-name style text', () => {
    const result = TextField({
      ...TextField.defaultProps,
      'class-name': 'style text',
    });

    const { className } = result.props;
    expect(className).toBe('style text ');
  });
});
