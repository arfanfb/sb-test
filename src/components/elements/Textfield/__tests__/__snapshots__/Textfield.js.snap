// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/textfield render 1`] = `
<input
  className=" "
  placeholder="Placeholder..."
  width="200"
/>
`;
